node/bft/src/helpers/dag.rs

```diff
@@ -113,19 +127,29 @@ impl<N: Network> DAG<N> {
         /* Updates */
 
         // Update the recently committed IDs.
-        self.recent_committed_ids.entry(certificate_round).or_default().insert(certificate_id);
+        let is_new = self.recent_committed_ids.entry(certificate_round).or_default().insert(certificate_id);
+        if !is_new {
+            //TODO (kaimast): return early here?
```

Comment:
Thank you!

Some open TODOs:
- PR message indicates "However, I want to perform another run of stresstest to be on the safe side.". Please indicate on a high level in the PR message which tests you ran.
- added @raychu86 for another independent strong review, maybe it would be good if you sit down together.
- @kaimast would you like to resolve the TODOs before or during review with Ray?